src/Validation/EntityChecker.php

```diff
@@ -38,15 +39,36 @@
     public function exists(mixed $value, string $role, ?string $field = null, bool $ignoreCase = false): bool
     {
         $repository = $this->orm->getRepository($role);
-        if ($field === null) {
-            return $repository->findByPK($value) !== null;
-        }
 
-        if ($ignoreCase && $repository instanceof Repository) {
-            return $this->addCaseInsensitiveWhere($repository->select(), $field, $value)->fetchOne() !== null;
+        if (!empty($value)) {
+            if (is_scalar($value)) {
+                if ($field === null) {
+                    return $repository->findByPK($value) !== null;
+                }
+
+                if ($ignoreCase && $repository instanceof Repository) {
+                    return $this->addCaseInsensitiveWhere($repository->select(), $field, $value)->fetchOne() !== null;
+                }
+
+                return $repository->findOne([$field => $value]) !== null;
+            }
+
+            if (\is_array($value) && $repository instanceof Repository) {
```

Comment:
i think we have to add new parameter like `list` because this workflow breaks the supporting for composite PK.

Or use values array as composite PK, but array of array as PK list